lib/core/refresh.js

```diff
@@ -0,0 +1,246 @@
+import jwtDecode, { InvalidTokenError } from 'jwt-decode'
+import { getProp } from './utilities'
+import ExpiredAuthSessionError from './includes/ExpiredAuthSessionError'
+
+export class RefreshController {
+  constructor (scheme) {
+    this.scheme = scheme
+    this.refreshPromise = null
+    this.refreshInterval = undefined
+  }
+
+  // Returns a promise which is resolved when refresh is completed
+  // Call this function when you intercept a request with an expired token.
+  // Multiple requests will be queued until the first has completed token refresh.
+  handleRefresh () {
+    // Another request has started refreshing the token, wait for it to complete
+    if (this.refreshPromise) {
+      return this.refreshPromise
+    }
+
+    return this._doRefresh()
+  }
+
+  _doRefresh () {
+    this.refreshPromise = new Promise(resolve => {
+      this.scheme._refreshToken().then(response => {
+        this.refreshPromise = null
+        resolve(response)
+      })
+    })
+
+    return this.refreshPromise
+  }
+
+  initializeAutomaticRefresh () {
+    // If auto refresh is enabled, schedule token refresh
+    if (this.scheme.options.autoRefresh.enable) {
+      this.initializeScheduledRefresh()
+    }
+
+    this.initializeRequestInterceptor()
+  }
+
+  // ---------------------------------------------------------------
+  // Refreshes token on set intervals
+  // Token is refreshed at 75% of the token expiration
+  // Call this function once from your mounted hook, client side
+  // ---------------------------------------------------------------
+  initializeScheduledRefresh () {
+    let intervalDuration = (this.scheme._getTokenExpiration() - Date.now()) * 0.75
+    if (intervalDuration < 1000) {
+      // in case you misconfigured refreshing this will save your auth-server from a self-induced DDoS-Attack
+      intervalDuration = 1000
+    }
+
+    this.refreshInterval = setInterval(() => {
```

Comment:
We need to guard it against `process.client` to avoid any unwanted SSR intervals.

Reply:
Ok!

Reply:
Fixed in commit https://github.com/nuxt-community/auth-module/pull/548/commits/eb38fd4f05cc717690a66c98088904861fca5b62